scipy/optimize/zeros.py

```diff
@@ -792,11 +792,15 @@ def brenth(f, a, b, args=(),
 
     A variation on the classic Brent routine to find a zero of the function f
     between the arguments a and b that uses hyperbolic extrapolation instead of
-    inverse quadratic extrapolation. There was a paper back in the 1980's ...
+    inverse quadratic extrapolation. Bus & Dekker (1975) guarantee convergence
+    for this method, claiming that the upper bound of function evaluations here
+    is 4 or 5 times lesser than that for bisection.
     f(a) and f(b) cannot have the same signs. Generally, on a par with the
     brent routine, but not as heavily tested. It is a safe version of the
-    secant method that uses hyperbolic extrapolation. The version here is by
-    Chuck Harris.
+    secant method that uses hyperbolic extrapolation.
+    The version here is by Chuck Harris, and implements Algorithm M of
+    [BusAndDekker1975]_, where further details (convergence properties,
+    additional remarks and such) can be found.
```

Comment:
I think we often put algorithm details in the `Notes` section, but maybe wait for another reviewer to suggest that before actually moving it. To be fair, there was already some algo discussion at this location.

Reply:
It's already in this place, so this PR extending that info to address the issue seems fine.